### PR TITLE
fix(condo): DOMA-9899 keep deeplink when changing sign in to register

### PR DESCRIPTION
**Pull request by tolmachev21**
Changes:
1) When pressing the registration or sign in button if there is next in the url, it remains

2) In order for sign in tab to switch to active, it was necessary to pass currentActiveKey with the next parameter

### CODE REVIEW

File: apps/condo/domains/common/components/HeaderActions.tsx [hunk 1]
```diff
@@ -16,6 +18,12 @@ export const TabsAuthAction: React.FC<ITabsActionsProps> = (props) => {
     const registerTab = intl.formatMessage({ id: 'pages.auth.RegistrationTitle' })
     const signInTab = intl.formatMessage({ id: 'pages.auth.SignInTitle' })
 
+    const router = useRouter()
+    const { query: { next }  } = router
+    const isValidNextUrl = next && !Array.isArray(next) && isSafeUrl(next)
+
+    console.log(next)
```
Comment (SavelevMatthew) [inline apps/condo/domains/common/components/HeaderActions.tsx:25]: ?

File: apps/condo/domains/common/components/HeaderActions.tsx [hunk 2]
```diff
@@ -24,8 +32,8 @@ export const TabsAuthAction: React.FC<ITabsActionsProps> = (props) => {
                 centered
                 animated={false}
             >
-                <Tabs.TabPane key='/auth/register?step=inputPhone' tab={registerTab}/>
-                <Tabs.TabPane key='/auth/signin' tab={signInTab}/>
+                <Tabs.TabPane key={isValidNextUrl ? `/auth/register?step=inputPhone&next=${next}` : '/auth/register?step=inputPhone'} tab={registerTab}/>
```
Comment (SavelevMatthew) [inline apps/condo/domains/common/components/HeaderActions.tsx:35]: You should encode URI component, next is plain text right now

File: apps/condo/pages/auth/signin.tsx [hunk 3]
```diff
@@ -13,12 +16,16 @@ const SignInPage: AuthPage = () => {
...
+    const { query: { next }  } = router
+    const isValidNextUrl = next && !Array.isArray(next) && isSafeUrl(next)
+
     return (
         <>
             <Head><title>{SignInTitleMsg}</title></Head>
             <Row justify='center'>
                 <Col span={16}>
-                    <TabsAuthAction currentActiveKey='/auth/signin'/>
+                    <TabsAuthAction currentActiveKey={isValidNextUrl ? `/auth/signin?next=${next}` : '/auth/signin'}/>
```
Comment (SavelevMatthew) [inline apps/condo/pages/auth/signin.tsx:28]: Same

File: apps/condo/domains/common/components/HeaderActions.tsx [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import { Tabs } from 'antd'
-import Router from 'next/router'
+import Router, { useRouter } from 'next/router'
 import React from 'react'
 
 import { useIntl } from '@open-condo/next/intl'
 
+import { isSafeUrl } from '@condo/domains/common/utils/url.utils'
```
Comment (sitozzz) [inline apps/condo/domains/common/components/HeaderActions.tsx:7]: Please setup automatic liner issues correction at the your IDE side

**Review by Alllex202 — COMMENTED** [review]
It would be nice to also pass a deep link for authorization via sbbol

File: apps/condo/domains/user/components/auth/InputPhoneForm.tsx [hunk 5]
```diff
@@ -22,6 +21,8 @@ import { START_CONFIRM_PHONE_MUTATION } from '@condo/domains/user/gql'
 
 
 import { RegisterContext } from './RegisterContextProvider'
+import { LoginWithSBBOLButton } from '../../../common/components/LoginWithSBBOLButton'
+import { isSafeUrl } from '../../../common/utils/url.utils'
```
Comment (Alllex202) [inline apps/condo/domains/user/components/auth/InputPhoneForm.tsx:25]: You should not use relative paths that have many hops to the parent directory.
Use alias `@condo/...`

File: apps/condo/domains/user/components/auth/InputPhoneForm.tsx [hunk 6]
```diff
@@ -62,6 +62,9 @@ export const InputPhoneForm: React.FC<IInputPhoneFormProps> = ({ onFinish }) =>
 
     const { publicRuntimeConfig: { hasSbbolAuth } } = getConfig()
 
+    const router = useRouter()
+    const { query: { next }  } = router
```
Comment (Alllex202) [inline apps/condo/domains/user/components/auth/InputPhoneForm.tsx:66]:
```suggestion
    const { query: { next } } = router
```

File: apps/condo/domains/common/components/HeaderActions.tsx [hunk 7]
```diff
@@ -16,16 +18,25 @@ export const TabsAuthAction: React.FC<ITabsActionsProps> = (props) => {
...
     return (
         <RemoveTabsLineWrapper>
             <Tabs
                 defaultActiveKey={currentActiveKey}
-                onChange={(activeKey) => Router.push(activeKey)}
+                onChange={(activeKey) => {
+                    if (activeKey === 'signin') {
+                        router.push(isValidNextUrl ? `/auth/signin?next=${encodeURIComponent(next)}` : '/auth/signin')
+                    } else if (activeKey === 'register') {
+                        router.push(isValidNextUrl ? `/auth/register?next=${encodeURIComponent(next)}` : '/auth/register')
```
Comment (Alllex202) [inline apps/condo/domains/common/components/HeaderActions.tsx:33]: Should we add `?step=inputPhone`?

File: apps/condo/domains/common/components/HeaderActions.tsx [hunk 8]
```diff
@@ -1,9 +1,11 @@
 import { Tabs } from 'antd'
-import Router from 'next/router'
+import { useRouter } from 'next/router'
 import React from 'react'
 
 import { useIntl } from '@open-condo/next/intl'
 
+import { isSafeUrl } from '@condo/domains/common/utils/url.utils'
```
Comment (Alllex202) [inline apps/condo/domains/common/components/HeaderActions.tsx:7]: Looks like you should set up autoformatting :)

Comment (tolmachev21) [reply apps/condo/domains/common/components/HeaderActions.tsx:7]: How I can do this?

File: apps/condo/domains/common/components/HeaderActions.tsx [hunk 9]
```diff
@@ -16,16 +17,25 @@ export const TabsAuthAction: React.FC<ITabsActionsProps> = (props) => {
...
     return (
         <RemoveTabsLineWrapper>
             <Tabs
                 defaultActiveKey={currentActiveKey}
-                onChange={(activeKey) => Router.push(activeKey)}
+                onChange={(activeKey) => {
+                    if (activeKey === 'signin') {
+                        router.push(isValidNextUrl ? `/auth/signin?next=${encodeURIComponent(next)}` : '/auth/signin')
+                    } else if (activeKey === 'register') {
+                        router.push(isValidNextUrl ? `/auth/register?step=inputPhone&next=${encodeURIComponent(next)}` : '/auth/register')
```
Comment (Alllex202) [inline apps/condo/domains/common/components/HeaderActions.tsx:32]: Should you add the "step" parameter here?

```suggestion
                        router.push(isValidNextUrl ? `/auth/register?step=inputPhone&next=${encodeURIComponent(next)}` : '/auth/register?step=inputPhone')
```

**Review by Alllex202 — APPROVED** [review]

**Review by nomerdvadcatpyat — APPROVED** [review]

**Pull request merged**